src/DotNetty.Codecs/Base64/Base64.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Base64
+{
+    using System;
+    using DotNetty.Buffers;
+
+    public class Base64
+    {
+        const int MAX_LINE_LENGTH = 76;
+        const byte EQUALS_SIGN = (byte)'='; //pad
+        const byte NEW_LINE = (byte)'\n';
+        const sbyte WHITE_SPACE_ENC = -5; // Indicates white space in encoding
+        const sbyte EQUALS_SIGN_ENC = -1; // Indicates equals sign in encoding
+
+        public static IByteBuffer Encode(IByteBuffer src)
+        {
+            return Encode(src, Base64Dialect.STANDARD);
+        }
+
+        public static IByteBuffer Encode(IByteBuffer src, Base64Dialect dialect)
+        {
+            return Encode(src, src.ReaderIndex, src.ReadableBytes, dialect.breakLinesByDefault, dialect);
+        }
+
+        public static IByteBuffer Encode(IByteBuffer src, bool breakLines, Base64Dialect dialect)
+        {
+            return Encode(src, src.ReaderIndex, src.ReadableBytes, breakLines, dialect);
+        }
+
+        public static IByteBuffer Encode(IByteBuffer src, int offset, int length, bool breakLines, Base64Dialect dialect)
+        {
+            return Encode(src, offset, length, breakLines, dialect, src.Allocator);
+        }
+
+        public unsafe static IByteBuffer Encode(IByteBuffer src, int offset, int length, bool breakLines, Base64Dialect dialect, IByteBufferAllocator alloctor)
```

Comment:
i correct this problem. check offset < src.ReaderIndex && offset + length > src.ReaderIndex + src.ReadableBytes now.